src/main/java/marquez/db/DatasetDao.java

```diff
@@ -57,9 +56,14 @@ void updateCurrentVersion(
   @SqlQuery("SELECT * FROM datasets WHERE urn = :urn.value")
   Optional<DatasetRow> findBy(@BindBean("urn") DatasetUrn urn);
 
-  @SqlQuery("SELECT * FROM datasets LIMIT :limit OFFSET :offset")
+  @SqlQuery(
+      "SELECT * "
+          + "FROM datasets d "
+          + "INNER JOIN namespaces n "
+          + "     ON (n.guid = d.namespace_guid AND n.name=:namespace)"
+          + "LIMIT :limit OFFSET :offset")
   List<DatasetRow> findAll(
-      @Bind("namespace") Namespace namespace,
```

Comment:
We can still use our defined types here using `@BindBean` instead `@Bind` (which only supports primitive types, `UUID`, etc)

Reply:
ah, right. switching to `@BindBean`